File: Kata 6/Consecutive strings/Consecutive strings.py

def longest_consec(strarr, k):
    if k > 0 and len(strarr) >= k:
        return ""
        
    result = ""    
    for index in range(len(strarr) - k + 1):
        s = ''.join(strarr[index:index+k])
        if len(s) > len(result):
            result = s
            
    return result
 
 
def longest_consec(strarr, k):
    if (len(strarr) == 0 or k <= 0) or len(strarr) < k:
        return ""
    lst = [''.join(strarr[i:i+k]) for i in range(len(strarr))]
    return max(lst, key= lambda x: len(x))

File: Kata 6/Consecutive strings/Consecutive strings.js
function longestConsec(strarr, k) {
    if (strarr.length == 0 || k<=0 ||  strarr.length <k)
        return ""
    var result = "";    
    for (var i = 0 ; i < strarr.length ;i++){
        var temp = strarr.slice(i,i + k).join("");
        result =   result.length >= temp.length ? result:temp
        
    }
    
    return result
}
